src/utilities/utility.ts

```diff
@@ -129,6 +137,75 @@ export function annotateChildPods(kubectl: Kubectl, resourceType: string, resour
     return commandExecutionResults;
 }
 
+export async function getFilePathsConfigs(): Promise<FileConfigPath> {
+
+    let filePathsConfig = <FileConfigPath>{};
+    const MANIFEST_PATHS_KEY = 'manifestFilePaths';
+    const HELM_CHART_KEY = 'helmChartFilePaths';
+    const DOCKERFILE_PATH_KEY = 'dockerfilePaths';
+    const DOCKERFILE_PATH_LABEL_KEY = 'dockerfile-path';
+
+    let inputManifestFiles = inputParams.manifests || [];
+    filePathsConfig[MANIFEST_PATHS_KEY] = inputManifestFiles;
+
+    let helmChartPaths = (process.env.HELM_CHART_PATHS && process.env.HELM_CHART_PATHS.split('\n').filter(path => path != "")) || [];
+    filePathsConfig[HELM_CHART_KEY] = helmChartPaths;
+
+    let imageNames = core.getInput('images').split('\n');
+    let imageDockerfilePathMap: any = {};
+    let pathValue: string, pathLink: string;
+    const branchOrTag: string = process.env.GITHUB_REF && process.env.GITHUB_REF.replace('refs/heads/','blob/').replace('refs/tags/','blob/');
+
+    //Fetching from image label if available
+    for (const image of imageNames) {
+        let args: string[] = [image];
+        let imageConfig: any;
+
+        try {
+            
+            await exec.exec('docker pull ', args, true).then(res => {
+                if (res.stderr != '' && !res.success) {
+                    throw new Error(`docker images pull failed with: ${res.stderr.match(/(.*)\s*$/)![0]}`);
+                }
+            });
+
+            await exec.exec('docker inspect --type=image', args, true).then(res => {
+                if (res.stderr != '' && !res.success) {
+                    throw new Error(`docker inspect call failed with: ${res.stderr.match(/(.*)\s*$/)![0]}`);
+                }
+
+                if (res.stdout) {
+                    imageConfig = JSON.parse(res.stdout);
+                }
+            });
+        }
+        catch (ex) {
+            core.warning(`Failed to get dockerfile paths for image ${image.toString()} | ` + ex);
+        }
+
+        if (imageConfig) {
+            imageConfig = imageConfig[0];
+            if ((imageConfig.Config) && (imageConfig.Config.Labels) && (imageConfig.Config.Labels[DOCKERFILE_PATH_LABEL_KEY])) {
+                pathValue = imageConfig.Config.Labels[DOCKERFILE_PATH_LABEL_KEY];
+                if(pathValue.startsWith('./')){  //if it is relative filepath convert to link from current repo
+                    pathLink = `https://github.com/${process.env.GITHUB_REPOSITORY}/${branchOrTag}/${pathValue}`;

```

Comment:
What if the image is built in a different repo?

Reply:
I have mentioned this in the README : 
`Use script to build image and add `dockerfile-path` label to it. The value expected is the link to the dockerfile. 
   If your dockerfile is in the same repo and branch where the workflow is run, it can be a relative path and it will be converted to a link for traceability.`

So, the the user is expected to give the correct file link if image is built in another repo.

Reply:
Ok. I don't think this approach is user-friendly. But if that's what it takes to ensure e2e traceability so be it!

Reply:
I agree, it is another step for the user, but for now (without something like Evidence store) we have to make do with this.